loopme-sdk/src/main/java/com/loopme/request/RequestValidator.kt

```diff
@@ -0,0 +1,138 @@
+package com.loopme.request
+
+import com.loopme.loaders.AdRequestType
+import com.loopme.request.RequestBuilder.*
+import org.json.JSONObject
+
+internal class RequestValidator {
+
+    private var isValid = true
+
+    /**
+     * After calling validateOrtbRequest function, this field should be populated. If empty hashmap, it means, that request is valid
+     */
+    val violations by lazy { HashMap<String, String>() }
```

Comment:
Hey @wnloopme, I wanted to bring up a quick question about mutable state in your validator class. Since `isValid` and `violations` are mutable state variables, they can be modified concurrently by multiple
calls to `validateOrtbRequest()` simultaneously. This could lead to unpredictable behavior and incorrect results. Were you aware of this potential issue?

To simplify the situation, it seems like a straightforward solution would be to simply return an array of `violations`. If this array is empty, it would imply that the request is valid. Conversely, if there
are violations in the array, you could use that information elsewhere as needed. I'd love to hear your thoughts on this matter.